sklearn/impute/_base.py

```diff
@@ -906,7 +906,7 @@ def _get_missing_features_info(self, X):
             imputer_mask.eliminate_zeros()
 
             if self.features == "missing-only":
-                n_missing = imputer_mask.getnnz(axis=0)
+                n_missing = count_nonzero(imputer_mask, axis=0)
```

Comment:
I prefer not to extend `count_nonzero` because we do not actually need `sample_weight` for scikit-learn's use case and only need `axis=0`.

Can we use `np.diff` here?

```suggestion
                n_missing = np.diff(imputer_mask.indptr)
```

Reply:
Not extending `count_nonzero` makes good sense to me too.  Thanks for stating that!

This sparse format is CSC so for `axis=0` we would need to use `np.bincount` instead of `np.diff` which does `axis=0` for CSR. But that is easy to implement.

I notice that the `sample_weight` argument is currently only used by CSR format and isn't included in the Array-API documents. Is it helpful for dense too?  It involves one-line or two-line numpy calls for sparse. I'm wondering if it might be good to support that in a different way than a `sparsefuncs.count_nonzero` keyword arg that doesn't match numpy or array-api. Maybe `weighted_count_nonzero()`.  But that is way beyond this PR. :)

I'll make this change. Thanks!